File: js/1719.js
const fs = require('fs');

const inputs = fs
  .readFileSync('example.txt')
  .toString()
  .trim()
  .split('\n')
  .map((e) => e.split(' ').map(Number));

const [n, m] = inputs[0];

const arr = Array(n + 1)
  .fill()
  .map((_, i) =>
    Array(n + 1)
      .fill()
      .map((e, j) => {
        if (i === j) return 0;
        return Infinity;
      })
  );

const ans = Array(n)
  .fill()
  .map((_, i) =>
    Array(n)
      .fill()
      .map((e, j) => {
        if (i === j) return '-';
        return j + 1;
      })
  );

for (let i = 1; i <= m; i++) {
  const [a, b, c] = inputs[i];
  arr[a][b] = c;
  arr[b][a] = c;
}

for (let k = 1; k <= n; k++) {
  for (let i = 1; i <= n; i++) {
    for (let j = 1; j <= n; j++) {
      if (arr[i][j] > arr[i][k] + arr[k][j]) {
        ans[i - 1][j - 1] = ans[i - 1][k - 1];
        arr[i][j] = arr[i][k] + arr[k][j];
      }
    }
  }
}

console.log(ans.map((e) => e.join(' ')).join('\n'));

File: c++/1719.cpp
#include <iostream>

#define INF 1e9

using namespace std;

int arr[201][201];
int ans[201][201];

int main() {
  freopen("example.txt", "r", stdin);
  int N, M;
  cin >> N >> M;

  for (int i = 1; i <= N; i++) {
    for (int j = 1; j <= N; j++) {
      if (i != j) {
        arr[i][j] = INF;
        ans[i][j] = j;
      }
    }
  }

  for (int i = 0; i < M; i++) {
    int a, b, c;
    cin >> a >> b >> c;
    arr[a][b] = c;
    arr[b][a] = c;
  }

  for (int k = 1; k <= N; k++) {
    for (int i = 1; i <= N; i++) {
      for (int j = 1; j <= N; j++) {
        if (arr[i][j] > arr[i][k] + arr[k][j]) {
          ans[i][j] = ans[i][k];
          arr[i][j] = arr[i][k] + arr[k][j];
        }
      }
    }
  }

  for (int i = 1; i <= N; i++) {
    for (int j = 1; j <= N; j++) {
      if (i == j)
        cout << '-' << ' ';
      else
        cout << ans[i][j] << ' ';
    }
    cout << '\n';
  }

  return 0;
}